docs/src/main/asciidoc/extension-maturity-model.adoc

```diff
@@ -0,0 +1,88 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+[id="extension-maturity-model"]
+= A maturity model for Quarkus extensions
+include::_attributes.adoc[]
+:diataxis-type: concept
+:categories: writing-extensions
+:topics: extensions
+:summary: Quarkus extensions can do a lot, or a little. This guide explains some of the capabilities extension authors might want to include.
+////
+The document header ends at the first blank line. Do not remove the blank line between the header and the abstract summary.
+////
+
+What makes a good Quarkus extension? What capabilities is a Quarkus extension expected to provide?
+
+image::extension-maturity-model.svg[A maturity model: Runtime actually works,Works in JVM mode,Works in dev mode, Works in native,Configuration support,Dev service,Dev UI, Use build-time application knowledge to eliminate wasteful runtime code paths (supersonic subatomic performance),Use build-time application knowledge to remove boilerplate,Codestart application template]
+
+The order in this model isn't exact. Different developers will have different views on what capabilities are most important. You may wish to (say) prioritise performance over enhancing your extension's Dev UI tile. That's fine!
+Also, not every step will apply to every extension. For example, you don't need a Dev Service if your extension doesn't depend on external services.
+It's completely OK to publish a first version of an extension that doesn't handle everything. In fact, it's OK if your extension _never_ gets to the more advanced features. This is a suggested pathway, not a minimum feature set.
+
+Also note that this list only includes the technical features of your extension.
+You might also want to think about how you share your extension, and how it presents itself to the world.
+The link:https://hub.quarkiverse.io/checklistfornewprojects/[new extension checklist] on the Quarkiverse Hub has a useful list of ways extensions can participate in the ecosystem.
```

Comment:
not blocking this pr for it but that link renders really badly for me. 

And the links on left on that page seem to be missing a lot compared to https://github.com/quarkiverse/quarkiverse/wiki/ChecklistForNewProjects - are these two using the same sources or split somehow?

Reply:
They're using the same source. The rendered version has the significant advantage that it's indexed by google, and thus exists to people trying to find the content. It's also got the advantages that the hosting isn't tied to github wikis, and we have a lot of control over the appearance (we just haven't invested the time to take advantage of that control).  

The navigation depth is different in the two renders, and I see there's some page width issues on a few rows. Is that the rendering issues you mean? Want to raise a defect against https://github.com/quarkiverse/quarkiverse for what you're seeing?

Reply:
sure - done, https://github.com/quarkiverse/quarkiverse/issues/288